recipes/libmagic/all/conanfile.py

```diff
@@ -0,0 +1,103 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import cross_building
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import copy, get, rm, rmdir, rename
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.layout import basic_layout
+import os
+
+required_conan_version = ">=1.52.0"
+
+class LibmagicConan(ConanFile):
+    name = "libmagic"
+    description = "Magic number recognition library - detect files with data in particular fixed formats."
+    license = "DocumentRef-COPYING:LicenseRef-BSD-2-Clause-File"  # Modified BSD 2-Clause that states no restrictions on US export
+    homepage = "https://github.com/file/file"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = "magic", "file format"
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("bzip2/1.0.8")
+        self.requires("lzip/1.23")
+        self.requires("xz_utils/5.4.4")
+        self.requires("zlib/[>=1.2.11 <2]")
+        self.requires("zstd/1.5.5")
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows is not supported yet")
+        if cross_building(self):
+            # TODO: Cross-compilation requires a build-context version of 'file' executable to be available.
+            # It might be possible to build the project for the 'build' platform first to make it work.
+            raise ConanInvalidConfiguration("Cross-compilation is not supported yet")
```

Comment:
This is very doable. When cross-building, add `libmagic` as a `tool_requires`. The Wayland recipe has to do something similar.

Reply:
Ok, sounds good. I'll give it a shot.

Reply:
And it worked. Nice! Thank you very much!